### PR TITLE
give error responses a status

### PR DESCRIPTION
**Pull request by NathanielBennett**
Correctly formats all error responses so that they match those previously served by the identity api: The format should be: 

```{"status":"error", "errors":[{"message":"xMessage","description":"xDescription"}]}```

### CODE REVIEW

File: src/main/scala/com/gu/sfl/controller/SaveForLaterController.scala [hunk 1]
```diff
@@ -6,13 +6,17 @@ import com.gu.sfl.model._
...
+  val missingAccessTokenResponse = lambdaErrorResponse(StatusCodes.forbidden, List(Error("Access denied", "could not find an access token.")))
+  val identityErrorResponse = lambdaErrorResponse(StatusCodes.internalServerError, List(Error("Access denied","Could not retrieve user id.")))
   val emptyArticlesResponse = LambdaResponse(StatusCodes.ok, Some(mapper.writeValueAsString(SavedArticles(List.empty))))
-  def maximumSavedArticlesErrorResponse(exception: Exception) = LambdaResponse(StatusCodes.entityTooLarge, Some(mapper.writeValueAsString(ErrorResponse(List(Error("Payload too large", exception.getMessage))))))
+
   def okSyncedPrefsResponse(syncedPrefs: SyncedPrefs): LambdaResponse = LambdaResponse(StatusCodes.ok, Some(mapper.writeValueAsString(SyncedPrefsResponse("ok", syncedPrefs))))
   def okSavedArticlesResponse(savedArticles: SavedArticles): LambdaResponse = LambdaResponse(StatusCodes.ok, Some(mapper.writeValueAsString(SavedArticlesResponse("ok", savedArticles))))
+  def serverErrorResponse(message: String) = lambdaErrorResponse(StatusCodes.internalServerError, List(Error("Server error.", message)))
```
Comment (alexduf) [inline src/main/scala/com/gu/sfl/controller/SaveForLaterController.scala]: [pedantic] I think the status donesn't need a full stop?

File: src/main/scala/com/gu/sfl/controller/FetchArticlesController.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ class FetchArticlesController(fetchSavedArticles: FetchSavedArticles)(implicit e
            case e: IdentityServiceException => Future.successful( identityErrorResponse )
            case e: MissingAccessTokenException => Future.successful( missingAccessTokenResponse )
            case e: UserNotFoundException => Future.successful(missingUserResponse)
-           case _ => Future.successful(serverErrorResponse )
+           case _ => Future.successful(serverErrorResponse(ex.getMessage) )
```
Comment (markrichardsg) [inline src/main/scala/com/gu/sfl/controller/FetchArticlesController.scala]: I'm not comfortable with returning any third party exception messages, as you don't always know what the message will include.
Is is safer to log with a UUID and return the UUID?

File: src/main/scala/com/gu/sfl/controller/SaveForLaterController.scala (hunk 1 above)
Comment (NathanielBennett) [reply src/main/scala/com/gu/sfl/controller/SaveForLaterController.scala]: Yeah. Thanks

File: src/main/scala/com/gu/sfl/controller/FetchArticlesController.scala (hunk 2 above)
Comment (alexduf) [reply src/main/scala/com/gu/sfl/controller/FetchArticlesController.scala]: Seen with Nathaniel: this is fixing a crash in prod so let's merge first and fix as a following PR

**Pull request merged**